src/ForkCMS/Utility/Csv/Writer.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace ForkCMS\Utility\Csv;
+
+use Backend\Core\Engine\Authentication;
+use Backend\Core\Engine\User;
+use PhpOffice\PhpSpreadsheet\IOFactory;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Writer\Csv;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class Writer
+{
+    private $charset;
+
+    private $options = [];
+
+    public function __construct(string $charset)
+    {
+        $this->charset = $charset;
+        $this->setDefaultOptions();
+    }
+
+    private function setDefaultOptions()
+    {
+        $this->options['Enclosure'] = '"';
+        $this->options['Delimiter'] = ',';
+        $this->options['LineEnding'] = "\n";
+        $this->options['UseBOM'] = true;
+    }
+
+    private function getWriter(Spreadsheet $spreadsheet): Csv
+    {
+        $writer = IOFactory::createWriter($spreadsheet, 'Csv');
+
+        if (!empty($this->options)) {
+            foreach ($this->options as $option => $value) {
+                $methodName = 'set' . $option;
+                if (method_exists($writer, $methodName)) {
+                    call_user_func([$writer, $methodName], $value);
+                }
+            }
+        }
+
+        return $writer;
+    }
+
+    public function forBackendUser(User $user): self
+    {
+        $this->options['Delimiter'] = $user->getSetting('csv_split_character');
```

Comment:
you need to cone the instance here and return the clone since using it after calling the forBackendUser will still run it with those changes otherwise

Reply:
fixed in f941db102a5cefc147be9ed7c784c88e8f6a7f73